### ISSUE TITLE
Indosyncracy of mean() on inflammation-01

### ISSUE THREAD

**Issue by erex** [swcarpentry/r-novice-inflammation#129]
This works as expected
```
# mean inflammation on day 7
mean(dat[, 7])
```
but this fails
```
# mean of measurements for patient 7
mean(dat[7,])
```
returns an error.  That is because there is no header in this file and arbitrary field names are give.

Might be worthwhile to note that it makes no medical sense to calculate the mean of 24 measurements.

**Comment by jdblischak** [comment]
Thanks for reviewing our lesson, @erex.

The issue you have found is an unfortunate artifact of how this lesson was developed. It is a translation of [novice-python-inflammation](https://github.com/swcarpentry/python-novice-inflammation).

When you slice a single column of a data frame, it returns a vector. When you slice a single row, it returns a data frame. The function `mean` does not work on a data frame (because a data frame often has data of different types). You can observe this behavior by running the following:

```r
class(mean(dat[, 7]))
class(mean(dat[7, ]))
```

The more natural way to store a bunch of numbers would be in R's `matrix` class. Then one could take the mean of a row. However, it was decided during the translation of this lesson that it would be better to use a data frame since it is so ubiquitous in R programming that novices should be exposed to it.

We have documented this [here](https://github.com/swcarpentry/r-novice-inflammation/blob/gh-pages/discussion.md#error-returned-when-attempting-to-calculate-mean-of-dataframe-rows) so that our instructors know about this quirk when teaching a workshop.

If you'd like to add a callout box to warn learners about this behavior, I'd welcome a Pull Request.

> Might be worthwhile to note that it makes no medical sense to calculate the mean of 24 measurements.

I feel that is a very strong opinion for such a context-dependent decision. The usefulness of a summary statistic depends on both the statistical properties of the data and the question being addressed.

**Comment by jdblischak** [comment]
@erex, are you interested in creating a callout box that warns learners about taking the mean of a row of a data frame?

**Comment by erex** [comment]
@jdblischak thanks for the invitation; I've got a fair bit on my plate for the next few days.  I'll try to do as you ask next week.  If I've not sent a pull request to you by the end of next week, I've gotten overtaken by events.

**Comment by dhaine** [comment]
Maybe with a mention about rowMeans()?

**Comment by lmweber** [comment]
Hi all, we had the same issue in a recent workshop. I have submitted a pull request with a Tip explaining the issue and suggesting use of as.numeric() to solve it: https://github.com/swcarpentry/r-novice-inflammation/pull/177

**Comment by chendaniely** [comment]
as @jdblischak mentioned, this is a result of translating from the Python lesson.

The R lesson was originally written using a matrix.  This allowed `mean()` to work in both directions.
There's was a discussion about using `Data.Frame` instead because that is `R`'s strong suit, and the reason why people use `R`.  When we switched the lesson to use `Data.Frame`, `mean` no longer worked row-wise.

We avoided mentioning/teaching `as.numeric` at that point in the lesson because we felt it would've been too much of a cognitive burden to new learners to also teach `class` and vector types at this point of the lesson
